internal/colorize/cropped.go

```diff
@@ -0,0 +1,58 @@
+package colorize
+
+type Entry struct {
+	Line   string
+	Length int
+}
+
+func GetCroppedText(text string, maxLen int) []Entry {
+	entries := make([]Entry, 0)
+	entryText := ""
+	pos := 0
+	count := 0
+	matches := colorRx.FindAllSubmatchIndex([]byte(text), -1)
+	runeText := []rune(text)
+
+	for pos < len(runeText) {
+		// If we reach an index that we recognize (ie. the start of a tag)
+		// then we write the whole tag, otherwise write by rune
+		for _, match := range matches {
+			start, stop := match[0], match[1]
+			if pos == start {
+				entryText += string(runeText[pos:stop])
+				pos = stop
+			}
+		}
+
+		if pos > len(runeText)-1 {
+			entries = append(entries, Entry{entryText, count})
+			break
+		}
+
+		if count == maxLen {
```

Comment:
You're not accounting for the color code on its own on a line anymore?

Reply:
With the changes you suggested above we don't end up with entries that are just colour tags. Though this _could_ happen if there is a newline character right before or after a colour tag but in that case it would be respecting the existing line ending. Do we want to do anything special/differently there?